### PR TITLE
[state-sync-v2] api get_first_txn_version

### PR DESCRIPTION
**Pull request by lightmark**
## Motivation

ready for state-sync v2. 
we need api to know the first available txn/txn_info version.

## Test Plan

ut

### CODE REVIEW

**Comment by libra-action** [comment]
**This PR may have modified JSON-RPC server code.**

Breaking changes policy:

1. Changing JSON-RPC method signatures or removing/modifying fields
   in the response in /V1 will not be accepted.
2. Adding fields is ok.
3. Adding a new JSON-RPC method is ok.

Please ensure you have also done the following to update the docs:

1. Update json-rpc/API-CHANGELOG.md.
2. Add/update type/method documentation under json-rpc/docs.

File: storage/diemdb/src/transaction_store/mod.rs [hunk 1]
```diff
@@ -116,6 +116,13 @@ impl TransactionStore {
         })
     }
 
+    /// Get the first version that txn starts existent.
+    pub fn get_first_txn_version(&self) -> Result<Option<Version>> {
+        let mut iter = self.db.iter::<TransactionSchema>(Default::default())?;
+        iter.seek_to_first();
+        iter.next().map(|res| res.map(|(v, _)| v)).transpose()
```
Comment (msmouse) [inline storage/diemdb/src/transaction_store/mod.rs:123]: The TxnInfo being smaller, is it fair to say that table can be faster to seek? :P

**Review by msmouse — COMMENTED** [review]
Please rebase on #9697 and remove the `unimplemented!()` everywhere..

Otherwise, looks good.

File: storage/diemdb/src/transaction_store/mod.rs (hunk 1 above)
Comment (lightmark) [reply storage/diemdb/src/transaction_store/mod.rs:123]: ...I don't think there is a big difference since it is binary search + linear at block level.

Comment (JoshLind) [reply storage/diemdb/src/transaction_store/mod.rs:123]: Yeah, I'm interested to know if this is true, too :D

**Review by JoshLind — DISMISSED** [review]
Looks good to me! 😄

File: storage/diemdb/src/transaction_store/mod.rs (hunk 1 above)
Comment (msmouse) [reply storage/diemdb/src/transaction_store/mod.rs:123]: Yeah no big deal. Virtually no difference at all. It's just a slightly larger kv being loaded and deserialized.

**Review by msmouse — DISMISSED** [review]

**Pull request closed without merging**